### PR TITLE
Enable sync on empty localstorage, force sync, sync on both load and reconnect.

### LINKED ISSUE #2278: Custom CSS only syncs across clients when they're open
* *Node version: 6.11.3* 
* *Browser version: many* 
* *Device, operating system: many* 
* *The Lounge version: 453eab3* 

---

Have one client open and change your custom CSS, then open another (like when you come home from work), your changes won't be there. 

Edit custom CSS while another client is open though and it gets the changes.

You'd expect it to work even when changes were made while other clients were closed.

### PR DESCRIPTION
**Pull request by creesch**
This fixes #2278

Turns out that I forgot to change `settings:get` to `setting:get` so settings never got synced on startup. Also made it so that the client will now sync settings when reconnecting. 

This also implements #2279 with the force sync button as discussed in that issue.

### CODE REVIEW

File: client/js/options.js [hunk 1]
```diff
@@ -300,8 +312,12 @@ function initialize() {
 		}
 	});
 
+	$settings.on("click", "#forceSync", () => {
```
Comment (xPaw) [inline client/js/options.js]: Should this enable sync?

Comment (creesch) [reply client/js/options.js]: I don't think so, now it has a single clear purpose. I personally wouldn't expect such a button to also enable sync.

And there might be use cases where people just want to force sync the settings of one client without enabling sync.

I also don't really expect buttons in an interface with check boxes to enable anything (Push notifications being a button seems odd to me for example). 

I thought about doing two checkboxes. One for enabling sync and one for force syncing before enabling sync but that seems confusing and easier to overlook than a big green button saying you can force sync. 

In short, this is the cleanest clearest solution in my opinion.

Comment (xPaw) [reply client/js/options.js]: Okay this is fine by me, I'll let other decide on this.

However can you change this line to read `$settings.find("#forceSync").on("click", () => {`

Comment (creesch) [reply client/js/options.js]: Done

File: client/views/windows/settings.tpl [hunk 2]
```diff
@@ -28,6 +28,10 @@
 			</label>
 			<p class="sync-warning-override"><strong>Warning</strong> Checking this box will override the settings of this client with those stored on the server.</p>
 			<p class="sync-warning-base"><strong>Warning</strong> No settings have been synced before. Enabling this will sync all settings of this client as the base for other clients.</p>
+			<label class="opt force-sync-button">
+				<button type="button" class="btn" id="forceSync">Force sync settings</button><br>
```
Comment (astorije) [inline client/views/windows/settings.tpl]: I feel like this is more commonly called "Set/Mark as default", but it's not perfect either. I just find "Force sync settings" not ideal.

I'm overall very confused by the wording of the feature, but I'm not sure how to make it clearer just now. Worth waiting for the feature to stabilize and see if we can improve.

screen shot 2018-04-02 at 02 22 24

Good thing it's an advanced feature :D

**Comment by astorije** [comment]
LGTM, but I know @McInkay has been pretty involved in this feature so it would be fair for him to give his opinion/approval before this makes it in :)

Thanks a lot for this, @creesch!

File: client/views/windows/settings.tpl (hunk 2 above)
Comment (creesch) [reply client/views/windows/settings.tpl]: I agree the wording isn't ideal but as you pointed out neither is other wording. And yeah this is a proper advanced setting only used for those people that already have settings they are particularly attached to.

**Comment by McInkay** [comment]
@astorije Do you want to review this anymore or are you happy for it to go in without you?

**Comment by astorije** [comment]
Nah, I trust both of you on this, and my UI/UX-hat-wearing concern I raised is shared by both you and @creesch so I know it's a "we're not sure how to do better for now" and not a "this is actually what we want", so we can iterate on that when we have better ideas.

**Comment by astorije** [comment]
Thanks a lot @creesch!
